src/systemctl/systemctl.c

```diff
@@ -2877,6 +2877,50 @@ static int start_unit_one(
                         return log_oom();
         }
 
+        r = sd_bus_get_property_trivial(
+                        bus,
+                        "org.freedesktop.systemd1",
+                        unit_path,
+                        "org.freedesktop.systemd1.Unit",
+                        "ConditionResult",
+                        error,
+                        SD_BUS_TYPE_BOOLEAN, &condition_met);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get property ConditionResult: %s", bus_error_message(error, r));
+
+        if (!condition_met) {
+                char *cond, *param;
+                int trigger, negate, state;
+
+                log_warning("The service you attempted to start was skipped:");
+
+                r = sd_bus_get_property(
+                                bus,
+                                "org.freedesktop.systemd1",
+                                unit_path,
+                                "org.freedesktop.systemd1.Unit",
+                                "Conditions",
+                                error,
+                                &reply,
+                                "a(sbbsi)");
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get property Conditions: %s", bus_error_message(error, r));
+
+                r = sd_bus_message_enter_container(reply, SD_BUS_TYPE_ARRAY, "(sbbsi)");
+                if (r < 0)
+                        return bus_log_parse_error(r);
+
+                while ((r = sd_bus_message_read(reply, "(sbbsi)", &cond, &trigger, &negate, &param, &state)) > 0)
+                        if (state < 0)
+                                /* The output here is the same as in print_status_info(),
+                                 * apart from the branch character. */
+                                log_warning("  %s=%s%s%s was not met\n",
+                                            cond,
+                                            trigger ? "|" : "",
+                                            negate ? "!" : "",
+                                            param);
+        }
```

Comment:
when iterating through an array, you should check the final return code, and handle it when it is an error (instead of 0). See other iteration loops, they all have a final if check for r.